### PR TITLE
Updates to agreements and memberships databases

### LINKED ISSUE #18: Add memberships database

### LINKED ISSUE #24: Recode End dates for treaties
End dates for treaties that are still in force should be distinguished from those where we do not know whether they are still in force or not. These treaties that are still in force should have End date coded as 9999-12-31. Any known future dates for termination (sunset clause) of the treaty should be noted down as the End date.

### LINKED ISSUE #25: Change Beg variable name to Begin

### LINKED ISSUE #26: Update stateIDs and state names in HUGGO_MEM
Align with updated countryregex

### LINKED ISSUE #28: add succession variable into HUGGO_MEM
Use succession variable to code rows where the unit had predecessors

### LINKED ISSUE #29: Add a variable to distinguish between formal agreements and WHO resolutions etc

### LINKED ISSUE #30: Add date of deposit as proxy for ratification dates where ratification dates are not available in HUGGO_MEM
As Diego has done in manyenviron, where specific dates of ratification for states are not available, we will code date of deposit as date of ratification in HUGGO_MEM for now. We are using the date of deposit as a proxy because some of the ratification dates recorded in sources such as the UN are also possibly deposit/notification dates (especially in the case of succession/accession). Note that this should be marked in an additional variable (see manytrade HUGGO_MEM).

### PR DESCRIPTION
**Pull request by jaeltan**
2024-03-13

# Description

This minor release includes updates to `agreements` and `memberships` databases, including updates to the coding in `HUGGO` dataset and the addition of a `HUGGO_MEM` dataset with membership information for the `agreements` in the agreements database.

## Package
* Added `memberships` to pkgdown.yml for website.
* Updated vignette with data from `memberships` datacube and used new graphing functions from `{manynet}`.

## Data

* Closed #25 by renaming `Beg` variable `Begin` across all datasets.
* Closed #29 by adding a `Formal` variable in `HUGGO` dataset to distinguish between formal agreements and other non-legally binding resolutions, declarations, etc.
* Closed #24 by recoding End dates for all agreements, so that agreements that are known to be still in force are coded '9999-12-31' rather than NA.
* Closed #18 by adding `HUGGO_MEM` dataset and memberships database.
* Closed #26 by updating the state names and stateIDs in the `HUGGO_MEM` dataset, so that stateIDs are unique and names are standardised.
* Closed #28 by adding a `Succession` variable in `HUGGO_MEM` to indicate the states that have a predecessor or successor since 1945.
* Added an `Accession` variable in `HUGGO_MEM` that records the date of accession to each agreement for states.
* Closed #30 by adding date of deposit as a proxy for date of ratification in variable `StateRat`.
* Added raw treaty texts in plain text format to HUGGO/TreatyTexts folder in the `agreements` database.

# Checklist:

- [x] PR form
  - [x] I have given this pull request an informative title
  - [x] Description above itemizes changes under subtitles, e.g. "## Collection""
  - [x] Any closed, fixed, or related issues are referenced and explained in the description above, e.g. "Fixed #0 by adding A"
  - [x] Package builds on my OS without issues
- [x] PR checks all pass for latest commit
  - [x] Package builds on Mac
  - [x] Package builds on Windows
  - [x] Package builds on Linux
- [x] Documentation
  - [x] Any new or modified functions or data have roxygen style documentation in their .R scripts
  - [x] Any longer functions are commented inline so that it is easier to debug in the future
  - [x] PR description above and the NEWS.md file are aligned
  - [x] DESCRIPTION file version is bumped by the appropriate increment (major, minor, patch)

### CODE REVIEW

File: R/manyhealth-agreements.R [hunk 1]
```diff
@@ -1,20 +1,63 @@
...
+#' @source
+#' \itemize{
+#' \item{WHO: }{
+#' D. Bhugra, S. Pathare, R. Joshi, et al.
+#' “A review of mental health policies from Commonwealthcountries”.
+#' In: _International Journal of Social Psychiatry_ 64.1 (2018), pp. 3-8.}
+#' \item{GHHR: }{
+#' B. M. Meier, O. A. Cabrera, A. Ayala, et al.
+#' “Bridging international law and rights-based litigation:mapping health-related rights through the development of the global health and human rights database”.
+#' In: _Health & Human Rights._ 14 (2012), p. 20.}
```
Comment (henriquesposito) [inline R/manyhealth-agreements.R:27]: Could you please just double check that `call_sources("manyhealth", "agreements")` works? I believe the ":" might not be understood correctly...

File: vignettes/healthdata.Rmd [hunk 2]
```diff
@@ -58,51 +76,64 @@ manyhealth::organizations$CHATHAM %>%
...
 references <- manyhealth::references$WHO_REF %>% 
   dplyr::distinct() %>%
-  dplyr::mutate(year = stringr::str_extract(Treaty2, "[:digit:]{4}")) %>%
+  dplyr::mutate(year = stringr::str_extract(Treaty1, "[:digit:]{4}")) %>%
   dplyr::filter(RefType == "Cites") %>%
-  dplyr::arrange(year)
+  dplyr::arrange(year) %>%
+  dplyr::filter(Treaty2 == "FRMWTC_2003A" | Treaty2 == "GLDPAH_2004S" | Treaty2 == "GLPCND_2000S")
 
 # Plot with migraph
```
Comment (henriquesposito) [inline vignettes/healthdata.Rmd:124]: manynet

**Review by henriquesposito — COMMENTED** [review]
Thank you so much for all the great work @jaeltan ! Just a few notes and questions on my side. Thank you.

File: vignettes/healthdata.Rmd (hunk 2 above)
Comment (jaeltan) [reply vignettes/healthdata.Rmd:124]: Thanks for spotting @henriquesposito !

File: R/manyhealth-agreements.R [hunk 3]
```diff
@@ -1,20 +1,63 @@
...
+#' \item{HUGGO: }{Hand-coded data by the GGO team}
+#' }
+#' @section Mapping:
+#' \itemize{
+#' \item{WHO: }{
+#' Variable Mapping
+#'
+#' |  *from*  | *to*
+#' |:------------:|:------------:|
+#' | NA | NA |
```
[the remark this answers is not in the document]
Comment (jaeltan) [reply R/manyhealth-agreements.R:48]: There were no original variable names as the data was scraped directly from the webpages, I could remove this for WHO instead?

File: R/manyhealth-agreements.R (hunk 1 above)
Comment (jaeltan) [reply R/manyhealth-agreements.R]: I've removed the 'In:' in the documentation.

File: vignettes/healthdata.Rmd [hunk 4]
```diff
@@ -16,35 +16,53 @@ knitr::opts_chunk$set(
...
+library(knitr)
+library(manynet)
+
+manyhealth::memberships$HUGGO_MEM %>%
+  dplyr::filter(messydates::year(Begin) > "1999" & messydates::year(Begin) < "2010") %>% 
+  dplyr::filter(stateID == "USA" | stateID == "CHN") %>%
+  as_tidygraph() %>%
+  mutate_ties(color = ifelse(to == 47, "China", "USA")) %>%
+  autographr(edge_color = "color")
+
```
[the remark this answers is not in the document]
Comment (jaeltan) [reply vignettes/healthdata.Rmd:61]: Vignette works now, it seems there was an issue with using as_tidygraph() on the full dataset so I changed the code to select only the variables being used for the plot.

File: R/manyhealth-agreements.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaeltan) [reply R/manyhealth-agreements.R:48]: Thanks @henriquesposito , yes it works now.

**Review by jhollway — APPROVED** [review]

**Review by henriquesposito — APPROVED** [review]
Thank you @jaeltan !

**Pull request merged**